standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -670,6 +725,8 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
 
       assert txnIds.size() == numTxns;
 
+      addTxnToMinHistoryLevel(dbConn, txnIds, minOpenTxnId);
```

Comment:
why not to put this under above useMinHistoryLevel check?

Reply:
That was my first intent, but it resulted in lock timeout, after inserting the new records in the txns table, the min open select was not running